src/megatron/bridge/training/setup.py

```diff
@@ -122,6 +122,8 @@ def setup(
     if cfg.comm_overlap is not None:
         cfg.comm_overlap.setup(cfg.model, cfg.optimizer, cfg.ddp)
 
+    cfg.validate()
+
```

Comment:
nit: you can move L113-125 into its own function, then unit test that the post-init of the subconfigs are run after the mixed precision and comm overlap runtime overrides are applied